File: app/src/main/java/com/mapbox/mapboxsdk/plugins/testapp/activity/TrafficActivity.java
package com.mapbox.mapboxsdk.plugins.testapp.activity;

import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.v7.app.AppCompatActivity;

import com.mapbox.mapboxsdk.maps.MapView;
import com.mapbox.mapboxsdk.maps.MapboxMap;
import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
import com.mapbox.mapboxsdk.plugins.testapp.R;
import com.mapbox.mapboxsdk.plugins.testapp.Utils;
import com.mapbox.mapboxsdk.plugins.traffic.TrafficPlugin;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import timber.log.Timber;

/**
 * Activity showcasing TrafficPlugin plugin integration
 */
public class TrafficActivity extends AppCompatActivity implements OnMapReadyCallback {

  @BindView(R.id.map_view)
  MapView mapView;

  @BindView(R.id.fabStyles)
  FloatingActionButton stylesFab;

  @BindView(R.id.fabTraffic)
  FloatingActionButton trafficFab;

  private MapboxMap mapboxMap;
  private TrafficPlugin trafficPlugin;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_traffic);
    ButterKnife.bind(this);

    mapView.setStyleUrl(Utils.getNextStyle());
    mapView.onCreate(savedInstanceState);
    mapView.getMapAsync(this);
  }

  @Override
  public void onMapReady(final MapboxMap mapboxMap) {
    this.mapboxMap = mapboxMap;
    this.trafficPlugin = new TrafficPlugin(mapView, mapboxMap);
    this.trafficPlugin.setVisibility(true); // Enable the traffic view by default
  }

  @OnClick(R.id.fabTraffic)
  public void onTrafficFabClick() {
    if (mapboxMap != null) {
      trafficPlugin.setVisibility(!trafficPlugin.isVisible());
      Timber.d("Traffic plugin is enabled : %s", trafficPlugin.isVisible());
    }
  }

  @OnClick(R.id.fabStyles)
  public void onStyleFabClick() {
    if (mapboxMap != null) {
      mapboxMap.setStyleUrl(Utils.getNextStyle());
    }
  }

  @Override
  protected void onStart() {
    super.onStart();
    mapView.onStart();
  }

  @Override
  protected void onResume() {
    super.onResume();
    mapView.onResume();
  }

  @Override
  protected void onPause() {
    super.onPause();
    mapView.onPause();
  }

  @Override
  protected void onStop() {
    super.onStop();
    mapView.onStop();
  }

  @Override
  protected void onSaveInstanceState(Bundle outState) {
    super.onSaveInstanceState(outState);
    mapView.onSaveInstanceState(outState);
  }

  @Override
  protected void onDestroy() {
    super.onDestroy();
    mapView.onDestroy();
  }

  @Override
  public void onLowMemory() {
    super.onLowMemory();
    mapView.onLowMemory();
  }

  public TrafficPlugin getTrafficPlugin() {
    return trafficPlugin;
  }
}

File: app/src/main/java/com/mapbox/mapboxsdk/plugins/testapp/activity/traffic/TrafficActivity.kt
package com.mapbox.mapboxsdk.plugins.testapp.activity.traffic

import android.os.Bundle
import android.support.v7.app.AppCompatActivity

import com.mapbox.mapboxsdk.maps.MapboxMap
import com.mapbox.mapboxsdk.maps.OnMapReadyCallback
import com.mapbox.mapboxsdk.maps.Style
import com.mapbox.mapboxsdk.plugins.testapp.R
import com.mapbox.mapboxsdk.plugins.testapp.Utils
import com.mapbox.mapboxsdk.plugins.traffic.TrafficPlugin

import kotlinx.android.synthetic.main.activity_traffic.*
import timber.log.Timber

/**
 * Activity showcasing TrafficPlugin plugin integration
 */
class TrafficActivity : AppCompatActivity(), OnMapReadyCallback {

    private var mapboxMap: MapboxMap? = null
    private var trafficPlugin: TrafficPlugin? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_traffic)
        mapView.onCreate(savedInstanceState)
        mapView.getMapAsync(this)

        fabTraffic.setOnClickListener { _ ->
            trafficPlugin?.let {
                it.setVisibility(!it.isVisible)
                Timber.d("Traffic plugin is enabled : %s", it.isVisible)
            }
        }

        fabStyles.setOnClickListener {
            mapboxMap?.setStyle(Style.Builder().fromUri(Utils.nextStyle))
        }
    }

    override fun onMapReady(mapboxMap: MapboxMap) {
        this.mapboxMap = mapboxMap
        mapboxMap.setStyle(Style.MAPBOX_STREETS) {
            this.trafficPlugin = TrafficPlugin(mapView, mapboxMap, it)
            this.trafficPlugin?.setVisibility(true) // Enable the traffic view by default
        }
    }

    override fun onStart() {
        super.onStart()
        mapView.onStart()
    }

    override fun onResume() {
        super.onResume()
        mapView.onResume()
    }

    override fun onPause() {
        super.onPause()
        mapView.onPause()
    }

    override fun onStop() {
        super.onStop()
        mapView.onStop()
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        mapView.onSaveInstanceState(outState)
    }

    override fun onDestroy() {
        super.onDestroy()
        mapView.onDestroy()
    }

    override fun onLowMemory() {
        super.onLowMemory()
        mapView.onLowMemory()
    }

    fun getTrafficPlugin(): TrafficPlugin? {
        return trafficPlugin
    }
}